### ISSUE TITLE
Error when using "object-curly-spacing" with ES6 export from default syntax

### ISSUE THREAD

**Issue by Bobsans** [eslint/eslint#13503]
**Environment Info:**
- Node version: v12.16.1
- npm version: v6.14.6
- Local ESLint version: vundefined (Currently used) (bug? installed: 7.4.0)
- Global ESLint version: Not found

**Parser:** 
- vue-eslint-parser
- babel-eslint

**Configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "root": true,
    "extends": ["eslint:recommended", "plugin:vue/recommended"],
    "parser": "vue-eslint-parser",
    "parserOptions": {
        "parser": "babel-eslint",
        "sourceType": "module",
        "ecmaVersion": 2020,
        "allowImportExportEverywhere": false
    },
    "env": {
        "browser": true,
        "es6": true,
        "node": true,
        "jasmine": true,
        "webextensions": true
    },
    "plugins": ["import", "promise", "standard", "vue"],
    "settings": {},
    "rules": {
        /* ------------------------------- */
        /* @@@@@ -= IMPORT PLUGIN =- @@@@@ */
        /* ------------------------------- */
        "import/default": "error",
        "import/dynamic-import-chunkname": [2, {"importFunctions": ["dynamicImport"], "webpackChunknameFormat": "[a-zA-Z0-57-9-/_]+"}],
        "import/export": "error",
        "import/exports-last": "warn",
        "import/extensions": ["warn", "never"],
        "import/first": "warn",
        "import/group-exports": ["off"],
        "import/max-dependencies": ["warn", {"max": 32}],
        "import/named": "warn",
        "import/namespace": "off",
        "import/newline-after-import": "warn",
        "import/no-absolute-path": "warn",
        "import/no-amd": "warn",
        "import/no-anonymous-default-export": "off",
        "import/no-commonjs": "off",
        "import/no-cycle": ["off"],
        "import/no-default-export": ["off"],
        "import/no-deprecated": "error",
        "import/no-duplicates": ["error", {"considerQueryString": true}],
        "import/no-dynamic-require": "error",
        "import/no-extraneous-dependencies": ["warn", {"devDependencies": false, "optionalDependencies": false, "peerDependencies": false}],
        "import/no-internal-modules": "off",
        "import/no-mutable-exports": "warn",
        "import/no-named-as-default-member": "warn",
        "import/no-named-as-default": "warn",
        "import/no-named-default": "warn",
        "import/no-named-export": "off",
        "import/no-namespace": "off",
        "import/no-nodejs-modules": ["warn", {"allow": ["path"]}],
        "import/no-relative-parent-imports": "off",
        "import/no-restricted-paths": "warn",
        "import/no-self-import": "warn",
        "import/no-unassigned-import": "off",
        "import/no-unresolved": "off",
        "import/no-unused-modules": "off",
        "import/no-useless-path-segments": ["warn", {"noUselessIndex": true}],
        "import/no-webpack-loader-syntax": "warn",
        "import/order": ["warn", {"groups": ["builtin", "internal", "external", "unknown", "parent", "sibling", "index"], "newlines-between": "ignore"}],
        "import/prefer-default-export": "off",
        "import/unambiguous": "off",
        /* -------------------------------- */
        /* @@@@@ -= PROMISE PLUGIN =- @@@@@ */
        /* -------------------------------- */
        "promise/always-return": "off",
        "promise/no-return-wrap": "error",
        "promise/param-names": "error",
        "promise/catch-or-return": "off",
        "promise/no-native": "off",
        "promise/no-nesting": "off",
        "promise/no-promise-in-callback": "warn",
        "promise/no-callback-in-promise": "off",
        "promise/avoid-new": "off",
        "promise/no-new-statics": "error",
        "promise/no-return-in-finally": "warn",
        "promise/valid-params": "warn",
        /* --------------------------------- */
        /* @@@@@ -= STANDARD PLUGIN =- @@@@@ */
        /* --------------------------------- */
        "standard/object-curly-even-spacing": ["error", "either"],
        "standard/array-bracket-even-spacing": ["error", "either"],
        "standard/computed-property-even-spacing": ["error", "even"],
        "standard/no-callback-literal": "error",
        /* ---------------------------- */
        /* @@@@@ -= VUE PLUGIN =- @@@@@ */
        /* ---------------------------- */

        /* -- Essential -- */
        "vue/no-async-in-computed-properties": "warn",
        "vue/no-dupe-keys": ["warn", {"groups": []}],
        "vue/no-duplicate-attributes": ["warn", {"allowCoexistClass": true, "allowCoexistStyle": true}],
        "vue/no-parsing-error": "error",
        "vue/no-reserved-keys": "warn",
        "vue/no-shared-component-data": "warn",
        "vue/no-side-effects-in-computed-properties": "warn",
        "vue/no-template-key": "error",
        "vue/no-textarea-mustache": "error",
        "vue/no-unused-components": "warn",
        "vue/no-unused-vars": "warn",
        "vue/no-use-v-if-with-v-for": ["warn", {"allowUsingIterationVar": true}],
        "vue/require-component-is": "warn",
        "vue/require-prop-type-constructor": "error",
        "vue/require-render-return": "error",
        "vue/require-v-for-key": "warn",
        "vue/require-valid-default-prop": "warn",
        "vue/return-in-computed-property": ["warn", {"treatUndefinedAsUnspecified": true}],
        "vue/use-v-on-exact": "warn",
        "vue/valid-template-root": "error",
        "vue/valid-v-bind": "error",
        "vue/valid-v-cloak": "error",
        "vue/valid-v-else-if": "error",
        "vue/valid-v-else": "error",
        "vue/valid-v-for": "error",
        "vue/valid-v-html": "error",
        "vue/valid-v-if": "error",
        "vue/valid-v-model": "error",
        "vue/valid-v-on": "error",
        "vue/valid-v-once": "error",
        "vue/valid-v-pre": "error",
        "vue/valid-v-show": "error",
        "vue/valid-v-text": "error",
        /* -- Strongly Recommended -- */
        "vue/attribute-hyphenation": ["error", "always"],
        "vue/html-closing-bracket-newline": ["error", {"singleline": "never", "multiline": "never"}],
        "vue/html-closing-bracket-spacing": ["warn", {"startTag": "never", "endTag": "never", "selfClosingTag": "never"}],
        "vue/html-end-tags": "error",
        "vue/html-indent": ["error", 4, {"attribute": 1, "baseIndent": 1, "closeBracket": 0, "alignAttributesVertically": true, "ignores": []}],
        "vue/html-quotes": ["error", "double"],
        "vue/html-self-closing": ["error", {"html": {"void": "never", "normal": "always", "component": "always"}, "svg": "always", "math": "always"}],
        "vue/max-attributes-per-line": ["warn", {"singleline": 32, "multiline": {"max": 1, "allowFirstLine": false}}],
        "vue/multiline-html-element-content-newline": ["warn", {"ignoreWhenEmpty": true, "ignores": ["pre", "textarea"], "allowEmptyLines": false}],
        "vue/mustache-interpolation-spacing": ["error", "always"],
        "vue/name-property-casing": ["error", "PascalCase"],
        "vue/no-multi-spaces": ["error", {"ignoreProperties": false}],
        "vue/no-spaces-around-equal-signs-in-attribute": "error",
        "vue/no-template-shadow": "error",
        "vue/prop-name-casing": ["error", "camelCase"],
        "vue/require-default-prop": "warn",
        "vue/require-prop-types": "warn",
        "vue/singleline-html-element-content-newline": ["off"],
        "vue/v-bind-style": ["warn", "shorthand"],
        "vue/v-on-style": ["warn", "shorthand"],
        /* -- Recommended -- */
        "vue/attributes-order": ["warn", {"order": [["CONDITIONALS", "RENDER_MODIFIERS", "LIST_RENDERING"], "DEFINITION", "UNIQUE", "TWO_WAY_BINDING", "OTHER_ATTR", "GLOBAL", "CONTENT", "OTHER_DIRECTIVES", "EVENTS"]}],
        "vue/no-v-html": "off",
        "vue/order-in-components": ["error", {"order": ["el", "name", "parent", "functional", ["delimiters", "comments"], "extends", "mixins", ["components", "directives", "filters"], "inheritAttrs", "model", ["props", "propsData"], "fetch", "asyncData", "data", "computed", "watch", "methods", "head", ["template", "render"], "renderError", "LIFECYCLE_HOOKS"]}],
        "vue/this-in-template": ["error", "never"],
        /* -- Uncategorized -- */
        "vue/component-name-in-template-casing": ["error", "kebab-case"],
        "vue/script-indent": ["warn", 4, {"baseIndent": 1, "switchCase": 1, "ignores": []}],
        "vue/v-on-function-call": ["off"],
        "vue/v-slot-style": ["error", {"atComponent": "shorthand", "default": "shorthand", "named": "shorthand"}],
        "vue/valid-v-bind-sync": ["error"],
        "vue/valid-v-slot": ["error"],
        /* ------------------------ */
        /* @@@@@ -= ESLINT =- @@@@@ */
        /* ------------------------ */
        "accessor-pairs": "error",
        "array-bracket-newline": ["error", "consistent"],
        "array-bracket-spacing": ["error", "never"],
        "array-callback-return": ["warn", {"allowImplicit": true, "checkForEach": false}],
        "array-element-newline": ["error", "consistent"],
        "arrow-body-style": "off",
        "arrow-parens": ["error", "always"],
        "arrow-spacing": ["error", {"before": true, "after": true}],
        "block-scoped-var": "error",
        "block-spacing": ["error", "always"],
        "brace-style": ["error", "1tbs", {"allowSingleLine": true}],
        "camelcase": ["warn", {"properties": "never", "ignoreDestructuring": true}],
        "capitalized-comments": "off",
        "class-methods-use-this": "off",
        "comma-dangle": ["warn", "never"],
        "comma-spacing": ["error", {"before": false, "after": true}],
        "comma-style": ["error", "last"],
        "complexity": "off",
        "computed-property-spacing": ["error", "never"],
        "consistent-return": "off",
        "consistent-this": ["warn", "self"],
        "constructor-super": "warn",
        "curly": ["error", "multi-line", "consistent"],
        "default-case": ["warn", {"commentPattern": "^no\\sdefault"}],
        "dot-location": ["error", "property"],
        "dot-notation": "error",
        "eol-last": ["error", "always"],
        "eqeqeq": ["error", "always", {"null": "ignore"}],
        "for-direction": "off",
        "func-call-spacing": ["error", "never"],
        "func-name-matching": ["warn", {"considerPropertyDescriptor": true}],
        "func-names": "off",
        "func-style": ["warn", "declaration", {"allowArrowFunctions": true}],
        "function-call-argument-newline": ["error", "never"],
        "function-paren-newline": ["error", "never"],
        "generator-star-spacing": ["error", {"before": false, "after": true, "anonymous": "neither", "method": {"before": true, "after": true}}],
        "getter-return": ["error", {"allowImplicit": true}],
        "guard-for-in": "off",
        "handle-callback-err": ["error", "^(err|error)$"],
        "id-length": "off",
        "id-match": "off",
        "implicit-arrow-linebreak": ["error", "beside"],
        "indent-legacy": ["error", 4, {"SwitchCase": 1, "MemberExpression": 1}],
        "indent": ["off", 4, {"SwitchCase": 1, "MemberExpression": 1}],
        "init-declarations": "off",
        "jsx-quotes": ["error", "prefer-double"],
        "key-spacing": ["error", {"beforeColon": false, "afterColon": true, "mode": "strict"}],
        "keyword-spacing": ["error", {"before": true, "after": true}],
        "line-comment-position": "off",
        "linebreak-style": ["warn", "unix"],
        "lines-around-comment": "off",
        "lines-around-directive": ["warn", {"before": "never", "after": "always"}],
        "lines-between-class-members": ["warn", "always"],
        "max-depth": ["warn", 10],
        "max-len": "off",
        "max-lines": "off",
        "max-nested-callbacks": ["warn", 10],
        "max-params": ["warn", 10],
        "max-statements-per-line": ["error", {"max": 1}],
        "max-statements": ["warn", 100, {"ignoreTopLevelFunctions": true}],
        "multiline-comment-style": "off",
        "multiline-ternary": ["error", "never"],
        "new-cap": ["error", {"newIsCap": true, "capIsNew": false}],
        "new-parens": ["error", "always"],
        "newline-after-var": "off",
        "newline-before-return": "off",
        "newline-per-chained-call": "off",
        "no-alert": "warn",
        "no-array-constructor": "error",
        "no-await-in-loop": "warn",
        "no-bitwise": ["warn", {"allow": ["~"], "int32Hint": true}],
        "no-buffer-constructor": "error",
        "no-caller": "error",
        "no-case-declarations": "error",
        "no-catch-shadow": "warn",
        "no-class-assign": "error",
        "no-compare-neg-zero": "error",
        "no-cond-assign": ["error", "except-parens"],
        "no-confusing-arrow": "off",
        "no-console": "warn",
        "no-const-assign": "error",
        "no-constant-condition": ["error", {"checkLoops": false}],
        "no-continue": "off",
        "no-control-regex": "error",
        "no-debugger": "error",
        "no-delete-var": "error",
        "no-div-regex": "error",
        "no-dupe-args": "error",
        "no-dupe-class-members": "error",
        "no-dupe-keys": "error",
        "no-duplicate-case": "error",
        "no-duplicate-imports": "error",
        "no-else-return": ["warn", {"allowElseIf": false}],
        "no-empty-character-class": "error",
        "no-empty-function": "error",
        "no-empty-pattern": "error",
        "no-empty": ["error", {"allowEmptyCatch": true}],
        "no-eq-null": "off",
        "no-eval": "error",
        "no-ex-assign": "error",
        "no-extend-native": "error",
        "no-extra-bind": "error",
        "no-extra-boolean-cast": "error",
        "no-extra-label": "error",
        "no-extra-parens": ["error", "functions"],
        "no-extra-semi": "error",
        "no-fallthrough": "off",
        "no-floating-decimal": "error",
        "no-func-assign": "error",
        "no-global-assign": "error",
        "no-implicit-coercion": "error",
        "no-implicit-globals": "error",
        "no-implied-eval": "error",
        "no-inline-comments": "off",
        "no-inner-declarations": ["error", "functions"],
        "no-invalid-regexp": "error",
        "no-invalid-this": "warn",
        "no-irregular-whitespace": "error",
        "no-iterator": "error",
        "no-label-var": "error",
        "no-labels": ["error", {"allowLoop": false, "allowSwitch": false}],
        "no-lone-blocks": "error",
        "no-lonely-if": "error",
        "no-loop-func": "warn",
        "no-magic-numbers": "off",
        "no-mixed-operators": ["error", {"groups": [["==", "!=", "===", "!==", ">", ">=", "<", "<="], ["&&", "||"], ["in", "instanceof"]], "allowSamePrecedence": true}],
        "no-mixed-requires": "error",
        "no-mixed-spaces-and-tabs": "error",
        "no-multi-assign": "off",
        "no-multi-spaces": "error",
        "no-multi-str": "error",
        "no-multiple-empty-lines": ["error", {"max": 1, "maxEOF": 0}],
        "no-native-reassign": "error",
        "no-negated-condition": "off",
        "no-negated-in-lhs": "error",
        "no-nested-ternary": "off",
        "no-new-func": "error",
        "no-new-object": "error",
        "no-new-require": "error",
        "no-new-symbol": "error",
        "no-new-wrappers": "error",
        "no-new": "off",
        "no-obj-calls": "error",
        "no-octal-escape": "error",
        "no-octal": "error",
        "no-param-reassign": "off",
        "no-path-concat": "error",
        "no-plusplus": "off",
        "no-process-env": "error",
        "no-process-exit": "error",
        "no-proto": "error",
        "no-prototype-builtins": "error",
        "no-redeclare": "error",
        "no-regex-spaces": "error",
        "no-restricted-globals": "off",
        "no-restricted-imports": "off",
        "no-restricted-modules": "off",
        "no-restricted-properties": "off",
        "no-restricted-syntax": "off",
        "no-return-assign": ["warn", "except-parens"],
        "no-return-await": "off",
        "no-script-url": "error",
        "no-self-assign": "error",
        "no-self-compare": "error",
        "no-sequences": "error",
        "no-shadow-restricted-names": "error",
        "no-shadow": "off",
        "no-sparse-arrays": "error",
        "no-sync": "off",
        "no-tabs": "error",
        "no-template-curly-in-string": "error",
        "no-ternary": "off",
        "no-this-before-super": "error",
        "no-throw-literal": "error",
        "no-trailing-spaces": "off",
        "no-undef-init": "error",
        "no-undef": "error",
        "no-undefined": "off",
        "no-underscore-dangle": "off",
        "no-unexpected-multiline": "error",
        "no-unmodified-loop-condition": "error",
        "no-unneeded-ternary": ["error", {"defaultAssignment": false}],
        "no-unreachable": "error",
        "no-unsafe-finally": "error",
        "no-unsafe-negation": "error",
        "no-unused-expressions": ["error", {"allowShortCircuit": true, "allowTernary": true, "allowTaggedTemplates": true}],
        "no-unused-labels": "error",
        "no-unused-vars": ["error", {"vars": "all", "args": "none", "ignoreRestSiblings": true, "caughtErrors": "all"}],
        "no-use-before-define": ["error", {"functions": false, "classes": false, "variables": false}],
        "no-useless-call": "error",
        "no-useless-computed-key": "error",
        "no-useless-concat": "error",
        "no-useless-constructor": "error",
        "no-useless-escape": "error",
        "no-useless-rename": "error",
        "no-useless-return": "error",
        "no-whitespace-before-property": "error",
        "no-var": "warn",
        "no-void": "error",
        "no-warning-comments": "off",
        "no-with": "error",
        "nonblock-statement-body-position": ["error", "below"],
        "object-curly-newline": ["error", {"multiline": true, "consistent": true}],
        "object-curly-spacing": ["error", "never"],
        "object-property-newline": ["error", {"allowMultiplePropertiesPerLine": true}],
        "object-shorthand": ["error", "always"],
        "one-var-declaration-per-line": "off",
        "one-var": "off",
        "operator-assignment": ["warn", "always"],
        "operator-linebreak": ["error", "after", {"overrides": {"?": "before", ":": "before"}}],
        "padded-blocks": ["error", {"blocks": "never", "switches": "never", "classes": "never"}],
        "padding-line-between-statements": "off",
        "prefer-arrow-callback": ["warn", {"allowNamedFunctions": false, "allowUnboundThis": false}],
        "prefer-const": ["warn", {"destructuring": "any", "ignoreReadBeforeAssign": false}],
        "prefer-destructuring": ["warn", {"object": true, "array": false}],
        "prefer-numeric-literals": "error",
        "prefer-promise-reject-errors": "error",
        "prefer-reflect": "off",
        "prefer-rest-params": "error",
        "prefer-spread": "error",
        "prefer-template": "error",
        "quote-props": ["warn", "as-needed", {"unnecessary": false}],
        "quotes": ["warn", "single", {"avoidEscape": true, "allowTemplateLiterals": true}],
        "radix": ["error", "as-needed"],
        "require-await": "error",
        "require-jsdoc": "off",
        "require-yield": "error",
        "rest-spread-spacing": ["error", "never"],
        "semi-spacing": ["error", {"before": false, "after": true}],
        "semi-style": ["error", "last"],
        "semi": ["warn", "always"],
        "sort-imports": "off",
        "sort-keys": "off",
        "sort-vars": "off",
        "space-before-blocks": ["error", "always"],
        "space-before-function-paren": ["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}],
        "space-in-parens": ["error", "never"],
        "space-infix-ops": ["error", {"int32Hint": true}],
        "space-unary-ops": ["error", {"words": true, "nonwords": false}],
        "spaced-comment": ["error", "always", {"line": {"markers": ["*package", "!", "/", ","]}, "block": {"balanced": true, "markers": ["*package", "!", ",", ":", "::", "flow-include"], "exceptions": ["*"]}}],
        "strict": "off",
        "switch-colon-spacing": ["error", {"before": false, "after": true}],
        "symbol-description": "error",
        "template-curly-spacing": ["error", "never"],
        "template-tag-spacing": ["error", "never"],
        "unicode-bom": ["error", "never"],
        "use-isnan": "error",
        "valid-jsdoc": "off",
        "valid-typeof": ["error", {"requireStringLiterals": true}],
        "vars-on-top": "off",
        "wrap-iife": ["error", "any", {"functionPrototypeMethods": true}],
        "wrap-regex": "off",
        "yield-star-spacing": ["error", "after"],
        "yoda": ["error", "never", {"exceptRange": true}]
    }
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export HelperMixin from './user-helper';
export AttributesMixin from './user-attributes';
```

```bash
eslint -f friendly --ext .js,.vue assets

```

**What did you expect to happen?**
Errors around `HelperMixin` and `AttributesMixin `: "ESLint: There should be no space before 'from'.(object-curly-spacing)"

**What actually happened? Please include the actual, raw output from ESLint.**
```text
There should be no space after 'export'

assets\management\mixins\index.js:1:7
   > 1 | export HelperMixin from './user-helper';
       |       ^
     2 | export AttributesMixin from './user-attributes';
     3 | 

✘  http://eslint.org/docs/rules/object-curly-spacing

There should be no space before 'from'

assets\management\mixins\index.js:1:34
   > 1 | export HelperMixin from './user-helper';
       |                   ^
     2 | export AttributesMixin from './user-attributes';
     3 | 

✘  http://eslint.org/docs/rules/object-curly-spacing

There should be no space after 'export'

assets\management\mixins\index.js:2:7
     1 | export HelperMixin from './user-helper';
   > 2 | export AttributesMixin from './user-attributes';
       |       ^
     3 | 

✘  http://eslint.org/docs/rules/object-curly-spacing

There should be no space before 'from'

assets\management\mixins\index.js:2:36
     1 | export HelperMixin from './user-helper';
   > 2 | export AttributesMixin from './user-attributes';
       |                       ^
     3 | 
```

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by anikethsaha** [comment]
Thanks for the report 👍 

I think eslint doesn't support default export as its underlining parser (Acorn, Espree) doesn't support that.

 `babel-eslint` does support that. 

So, I would recommend to turn off the base rule and add this rule from `eslint-plugin-babel` ([ref this](https://github.com/babel/eslint-plugin-babel/blob/master/rules/object-curly-spacing.js))

I think they do support default export in the plugin [ref this test](https://github.com/babel/eslint-plugin-babel/blob/master/tests/rules/object-curly-spacing.js#L145)

**Comment by Bobsans** [comment]
Yes. It's work for me. Thanks.
